indexing-service/src/main/java/org/apache/druid/indexing/worker/config/WorkerConfig.java

```diff
@@ -33,30 +34,32 @@
 {
   @JsonProperty
   @NotNull
-  private final String ip = DruidNode.getDefaultHost();
+  private String ip = DruidNode.getDefaultHost();
 
   @JsonProperty
   @NotNull
-  private final String version = "0";
+  private String version = "0";
 
   @JsonProperty
   @Min(1)
-  private final int capacity = Math.max(1, JvmUtils.getRuntimeInfo().getAvailableProcessors() - 1);
+  private int capacity = Math.max(1, JvmUtils.getRuntimeInfo().getAvailableProcessors() - 1);
 
   @JsonProperty
-  private final long intermediaryPartitionDiscoveryPeriodSec = 60L;
+  private long intermediaryPartitionDiscoveryPeriodSec = 60L;
 
   @JsonProperty
-  private final long intermediaryPartitionCleanupPeriodSec = 300L;
+  private long intermediaryPartitionCleanupPeriodSec = 300L;
 
   @JsonProperty
-  private final Period intermediaryPartitionTimeout = new Period("P1D");
+  private Period intermediaryPartitionTimeout = new Period("P1D");
 
+  @Nonnull
```

Comment:
Is this necessary? In general, we add `@EverythingIsNonnullByDefault` on the package declaration in `package-info.java` instead and assume that everything that is not explicitly annotated `@Nullable` is non-null. (Actually, it's violated in a lot of places, and it' a many years-long quest to eliminate those discrepancies, but still.)

`javax.validation.constraints.NotNull` on json properties is something different.

Reply:
Thanks for the response. 

The `@Nonnull` it isn't necessary of course. Right now I don't see any `package-info.java` in the indexing part, as `find indexing-service/src/ | grep -i package-info.java` doesn't give anything back.

I've removed the annotation for now. I'll create different PR for adding the `@EverythingIsNonnullByDefault` on the indexing part. Still a lot to be done, but the spotbugs is closing in on the voilations. I'm trying to get rid of all the voilations in an interative way by solving the issues one by one, for example in https://github.com/apache/incubator-druid/pull/8129/